File: solutions/python/grade-school/1/grade_school.py
class School:
    def __init__(self):
        self._grades = {}
        self._added = []

    def add_student(self, name, grade):
        for students in self._grades.values():
            if name in students:
                self._added.append(False)
                return
        self._grades.setdefault(grade, []).append(name)
        self._added.append(True)

    def roster(self):
        return [
            student
            for grade in sorted(self._grades)
            for student in sorted(self._grades[grade])
        ]

    def grade(self, grade_number):
        return sorted(self._grades.get(grade_number, []))

    def added(self):
        return self._added

File: solutions/javascript/grade-school/1/grade-school.js
export class GradeSchool {
  constructor() {
    this.grades = new Map();
  }

  add(name, grade) {
    const allStudents = this.roster();
    if (allStudents.includes(name)) {
      return false;
    }
    if (!this.grades.has(grade)) {
      this.grades.set(grade, []);
    }
    this.grades.get(grade).push(name);
    return true;
  }

  grade(grade) {
    const students = this.grades.get(grade) || [];
    return [...students].sort();
  }

  roster() {
    const sortedGrades = [...this.grades.keys()].sort((a, b) => a - b);
    return sortedGrades.flatMap((g) => this.grade(g));
  }
}